ofctrl/fgraphFlow.go

```diff
@@ -774,6 +775,32 @@ func (self *Flow) installFlowActions(flowMod *openflow13.FlowMod,
 
 			log.Debugf("flow install. Added popVlan action: %+v", popVlan)
 
+		case ActTypePushMpls:
+			// Create push mpls action
+			pushMpls := openflow13.NewActionPushMpls(flowAction.mplsEtherType)
+
+			// Add it to instruction
+			err = actInstr.AddAction(pushMpls, true)
+			if err != nil {
+				return err
+			}
+			addActn = true
+
+			log.Debugf("flow install. Added pushMpls action: %+v", pushMpls)
+
+		case ActTypePopMpls:
+			// Create pop mpls action
+			popMpls := openflow13.NewActionPopMpls(flowAction.mplsEtherType)
```

Comment:
Could use action `PopMPLSAction` here to avoid duplicate code? e.g. `popMpls := PopMPLSAction{EtherType: flowAction.mplsEtherType}.GetActionMessage()

Reply:
it worked as usual，could please review it again? Thanks !